File: android/src/main/java/com/rnmapbox/rnmbx/components/styles/light/RNMBXLightManager.java
package com.rnmapbox.rnmbx.components.styles.light;

import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.ViewGroupManager;
import com.facebook.react.uimanager.annotations.ReactProp;

public class RNMBXLightManager extends ViewGroupManager<RNMBXLight> {
    public static final String REACT_CLASS = "RNMBXLight";

    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @Override
    protected RNMBXLight createViewInstance(ThemedReactContext reactContext) {
        return new RNMBXLight(reactContext);
    }

    @ReactProp(name="reactStyle")
    public void setReactStyle(RNMBXLight light, ReadableMap reactStyle) {
        light.setReactStyle(reactStyle);
    }
}

File: android/src/main/java/com/rnmapbox/rnmbx/components/styles/light/RNMBXLightManager.kt
package com.rnmapbox.rnmbx.components.styles.light

import com.facebook.react.bridge.Dynamic
import com.facebook.react.uimanager.ThemedReactContext
import com.rnmapbox.rnmbx.utils.extensions.asMapOrNull
import com.facebook.react.uimanager.ViewGroupManager
import com.facebook.react.uimanager.annotations.ReactProp
import com.facebook.react.viewmanagers.RNMBXLightManagerInterface
import com.facebook.react.uimanager.ViewManagerDelegate
import com.facebook.react.viewmanagers.RNMBXLightManagerDelegate

class RNMBXLightManager : ViewGroupManager<RNMBXLight>(),
    RNMBXLightManagerInterface<RNMBXLight> {

    private val delegate = RNMBXLightManagerDelegate<RNMBXLight, RNMBXLightManager>(this)

    override fun getDelegate(): ViewManagerDelegate<RNMBXLight> = delegate
    override fun getName(): String {
        return REACT_CLASS
    }

    override fun createViewInstance(reactContext: ThemedReactContext): RNMBXLight {
        return RNMBXLight(reactContext)
    }

    @ReactProp(name = "reactStyle")
    override fun setReactStyle(light: RNMBXLight, reactStyle: Dynamic) {
        light.setReactStyle(reactStyle.asMapOrNull())
    }

    companion object {
        const val REACT_CLASS = "RNMBXLight"
    }
}